oceanprefilter/utils.go

```diff
@@ -99,6 +100,79 @@ func splitUpImage(img image.Image, exZone *image.Rectangle, yValue, nh, nv int)
 			}
 			bandImg := image.NewRGBA(bandRect)
 			draw.Draw(bandImg, bandImg.Bounds(), croppedImg, image.Point{bandRect.Min.X, bandRect.Min.Y + yValue}, draw.Src)
+
+			images = append(images, bandImg)
+		}
+	}
+	return images, nil
+}
+
+
+// crop the image from yValue -> img.Bounds().Max.Y
+// and then split the cropped image into nh horizontal and nv vertical bands of equal height and width (dimensions given)
+func splitUpImageConst(img image.Image, exZone *image.Rectangle, yValue, h, w int) ([]image.Image, error) {
+	if img == nil {
+		return nil, errors.New("input image to split up is nil")
+	}
+	if h <= 0 {
+		return nil, fmt.Errorf("height must be greater than 0")
+	}
+	if w <= 0 {
+		return nil, fmt.Errorf("width must be greater than 0")
+	}
```

Comment:
Don't use fmt.ErrorF if you're not providing a value in the string. either use `errors.New` if its just text, or use `errors.Errorf("height must be greater than 0, got %v", h)` to provide what the erroring value was